src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/R2RReader.cs

```diff
@@ -228,18 +228,15 @@ private unsafe void Initialize()
             {
                 Image = File.ReadAllBytes(Filename);
```

Comment:
There are more efficient ways to open PEReader on top of a file. (E.g. look at what the compiler does to open IL files). Can we use one of those?

Reply:
Do you mean using a `MemoryMappedViewAccessor` as in [here](https://github.com/dotnet/runtime/blob/f1c7ed882a90104ab7171c95f4767fc15bb50cd6/src/coreclr/src/tools/Common/Compiler/CompilerTypeSystemContext.cs#L158)?

That doesn't seem applicable here (or requires a lot of refactoring) because the `Image` property is used widely within `R2RDump`, at the end of `R2RDump`, we would have read a significant portion of the file anyway.

In the case of ILSpy, we won't hit this code path. ILSpy handed us an already loaded instance of PEReader and MetadataReader. I am hoping to eliminate [this](https://github.com/dotnet/runtime/blob/f1c7ed882a90104ab7171c95f4767fc15bb50cd6/src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/R2RReader.cs#L248) copy, but I hesitated after looking at how many times the code depended on `Image` is a raw array.

Reply:
Ok. For situation like this, you may consider converting between `ImmutableArray<byte>` and readonly `byte[]` using `Unsafe.As`. E.g. `Unsafe.As<byte[], ImmutableArray<byte>>(ref image))`.

Reply:
We had API proposal to add Marshal API for the unsafe `Immutable<T>` / `T[]` conversions, but the proposal was rejected with recommendation to use unsafe code.